staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/streaming_test.go

```diff
@@ -0,0 +1,318 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package json
+
+import (
+	"bytes"
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	fuzz "github.com/google/gofuzz"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	testapigroupv1 "k8s.io/apimachinery/pkg/apis/testapigroup/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func TestStreamingCollectionsEncoding(t *testing.T) {
```

Comment:
Do we need to test the failure case for an invalid list object with more than 3 fields?

```
type InvalidCarpList struct {
    metav1.TypeMeta   `json:",inline"`
    metav1.ListMeta   `json:"metadata,omitempty"`
    Items             []testapigroupv1.Carp `json:"items"`
    ExtraInformation  string                `json:"extra,omitempty"`
}
```

Reply:
Done